lib/voice/VoiceConnection.js

```diff
@@ -944,14 +1103,16 @@ class VoiceConnection extends EventEmitter {
         this.current.timeout = setTimeout(this.#send, this.current.startTime + this.current.pausedTime + this.current.playTime - Date.now());
     }
 
-    _sendAudioFrame(frame) {
+    _sendAudioFrame(unencryptedFrame) {
         const headerSize = this.sendHeader.length;
         this.sendHeader.writeUInt16BE(this.sequence, 2);
         this.sendHeader.writeUInt32BE(this.timestamp, 4);
 
         this.#nonce = (this.#nonce + 1) >>> 0;
         this.#nonceBuffer.writeUInt32BE(this.#nonce, 0);
 
+        const frame = this.#daveReady() && unencryptedFrame !== SILENCE_FRAME ? this.daveSession.encryptOpus(unencryptedFrame) : unencryptedFrame;
```

Comment:
a foreign buffer might be passed here (through `sendAudioFrame`), compare buffer contents instead

```suggestion
        const frame = this.#daveReady() && !unencryptedFrame.equals(SILENCE_FRAME) ? this.daveSession.encryptOpus(unencryptedFrame) : unencryptedFrame;
```